force-app/main/default/classes/SOQL.cls

```diff
@@ -85,9 +100,12 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static Map<Id, Task> tasksDueThisWeek () {
+        Date sevenDaysAway = System.today().addDays(7);
+        List<Task> listOfTasks = [SELECT Id, Subject FROM Task WHERE ActivityDate <= :sevenDaysAway];
+        Map<Id, Task> tasksById = new Map<Id, Task>(listOfTasks);
+        Return tasksById;
```

Comment:
Take a look at some of the key words for dates you can use in SOQL, like TODAY, TOMORROW, or some that you can modify like LAST_N_DAYS:n